### PR TITLE
FileEventMSEventsUpdate

### PR DESCRIPTION
**Pull request by sparkmark**
Required items, please complete
   
   Change(s):
   - Updated parser to include parsing for eventid's 5140 and 4670

   Reason for Change(s):
   - Enhancement of the existing windows file parser

   Version Updated:
   - Yes

   Testing Completed:
   - Tested in our lab

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by sparkmark** [comment]
@microsoft-github-policy-service agree company="AXA"

**Comment by v-atulyadav** [comment]
Hi @sparkmark, we will provide you an update on this shortly. Thanks

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml [hunk 1]
```diff
@@ -34,66 +70,179 @@ ParserQuery: |
...
+        | extend EventSchema = "FileEvent"
+            , EventSchemaVersion = "0.1.1"
+            , EventResult = "Success"
+            , EventCount = int(1)
+            , EventVendor = 'Microsoft'
+            , EventProduct = 'Security Events'
+            , Dvc = DvcHostname
+            , ActorWindowsUsername = ActorUsername
+            , User = ActorUsername
+            , ActorUserSid = ActorUserId
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml:246]: Please have a look at the data test results:
image

**Review by vakohl — CHANGES_REQUESTED** [review]

**Comment by v-atulyadav** [comment]
Hi @sparkmark, hope you are doing well. Just wanted to check if you got a chance to look at the @vakohl's suggestions. Please feel free to reach out to us for any queries and/or support.
Thanks

**Comment by sparkmark** [comment]
> Hi @sparkmark, hope you are doing well. Just wanted to check if you got a chance to look at the @vakohl's suggestions. Please feel free to reach out to us for any queries and/or support.
> Thanks

Hi @vakohl and @v-atulyadav ... I'm finding this somewhat frustrating. When we initially started this I was asked to separate all the events. Also we copied what had already been done in terms of fields being used and fields being projected away. But it would seem that the initial parser created didn't actually meet all the requirements you are now stipulating.

**Comment by vakohl** [comment]
@v-atulyadav I'm looking into past discussions internally regarding this PR. Pls wait for my update

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml [hunk 2]
```diff
@@ -13,12 +13,48 @@ References:
...
 Description: |
-  This ASIM parser supports normalizing Microsoft Windows Events (WindowsEvent and SecurityEvent tables) to the ASIM File Event normalized schema. Event IDs which are parsed as part of this parser: 4663
+  This ASIM parser supports normalizing Microsoft Windows Events (WindowsEvent and SecurityEvent tables) to the ASIM File Event normalized schema. Event IDs which are parsed as part of this parser: 4663, 5140, 4670
 ParserName: vimFileEventMicrosoftWindowsEvents
 ParserQuery: |
   let Parser=()
-  {
-  let EventTypeLookup = datatable (AccessMask:string,EventType:string)
+  {  
+  let EventType5140Lookup = datatable (AccessMask:string,EventType:string)
```
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml:21]: for now, we'll go with Mutiple tables. Once this PR is merged, we'll evaluate the parser to see if we can club these tables to one or keep them separated.

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml [hunk 3]
```diff
@@ -13,12 +13,48 @@ References:
...
+      , "%%4421", "N/A"
+      , "%%4422", "N/A"
+      , "%%4423", "N/A"
+      , "%%4424", "N/A"
+      , "%%1537", "N/A"
+      , "%%1538", "N/A"
+      , "%%1539", "N/A"
+      , "%%1540", "N/A"
+      , "%%1541", "N/A"
+      , "%%1542", "N/A"
```
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml:55]: Please use 'Other' if event subtype doesn't match with allowed values

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml [hunk 4]
```diff
@@ -13,12 +13,48 @@ References:
...
+  ];
+  let EventType4663Lookup = datatable (AccessMask:string,EventType:string)
```
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml:57]: for now, we'll go with Mutiple tables. Once this PR is merged, we'll evaluate the parser to see if we can club these tables to one or keep them separated.

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml [hunk 5]
```diff
@@ -34,66 +70,179 @@ ParserQuery: |
...
-    , EventSchemaVersion = "0.1.1"
-    , EventResult = "Success"
-    , EventCount = int(1)
-    , EventVendor = 'Microsoft'
-    , EventProduct = 'Security Events'
-    , Dvc = DvcHostname
-    , ActorWindowsUsername = ActorUsername
-    , User = ActorUsername
-    , ActorUserSid = ActorUserId
+  let 4663events =
```
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml:84]: We will evaluate this on our side.

**Comment by vakohl** [comment]
@sparkmark I've have resolved the comments where I have asked for merging of tables. Please have a look at the comments, let me know if you got any question.

**Comment by v-atulyadav** [comment]
Hi @sparkmark, please see the comments from @vakohl above. Thanks

**Comment by vakohl** [comment]
@sparkmark Wanted to check regarding this PR. Do you like to us to review the changes in this PR?

**Comment by v-atulyadav** [comment]
Hi @sparkmark,
Please confirm if you would like us to review the changes in this PR as per vakohl's comment? Thanks

**Comment by v-atulyadav** [comment]
Hi @sparkmark, 
We look forward to hearing back from you. Thanks

**Comment by v-atulyadav** [comment]
Hi @sparkmark,
We are closing this PR now if you still need support for this issue, you can re-open the PR at any time.
If you do re-open, we simply request that you ensure the PR has response to the last request. Thank you for your cooperation. Thank you.

**Pull request closed without merging**